Keen/Query/TimeframeConverter.cs

```diff
@@ -0,0 +1,48 @@
+﻿using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using System;
+
+
+namespace Keen.Core.Query
+{
+    /// <summary>
+    /// Used for converting IQueryTimeframe instances to/from JSON.
+    /// </summary>
+    internal class TimeframeConverter : JsonConverter
+    {
+        public override bool CanConvert(Type objectType)
+        {
+            return typeof(IQueryTimeframe).IsAssignableFrom(objectType);
+        }
+
+        public override void WriteJson(JsonWriter writer, object value, JsonSerializer serializer)
+        {
+            // QueryAbsoluteTimeframe has fields with JsonProperty attributes, so shouldn't need
```

Comment:
Yeah this is where I wish `JsonConverter`'s `CanWrite` were a method with a type parameter like `CanConvert()` or `CanConvert` were to take a direction parameter...but oh well. I'll throw to be clearer.